Does not build any more

This action odes not work any more:
```
#11 [builder 6/6] RUN make
  #11 0.161 make[1]: Entering directory '/tmp/cpdfsqueeze'
  #11 0.173 ocamlfind ocamldep cpdfsqueeze.ml > ._d/cpdfsqueeze.d
  #11 0.192 ocamlfind ocamlopt -package camlpdf -c -annot -bin-annot -g -safe-string -w -3 cpdfsqueeze.ml
  #11 0.227 File "cpdfsqueeze.ml", line 315, characters 108-120:
  #11 0.227 315 |       Pdfwrite.pdf_to_file_options ~recrypt:(Some best_password) ~generate_objstm:true false None false pdf !output_file
  #11 0.227                                                                                                                   ^^^^^^^^^^^^
  #11 0.227 Error: The function applied to this argument has type
  #11 0.227          ?preserve_objstm:bool -> ?compress_objstm:bool -> unit
  #11 0.227 This argument cannot be applied without label
  #11 0.229 make[1]: *** [OCamlMakefile:1076: cpdfsqueeze.cmx] Error 2
  #11 0.229 make[1]: Leaving directory '/tmp/cpdfsqueeze'
  #11 0.230 make: *** [OCamlMakefile:777: native-code] Error 2
  #11 ERROR: process "/bin/sh -c make" did not complete successfully: exit code: 2
```
I suppose the reason is that `debian:stable-slim` is used in `Dockerfile` and last saturday Debian Trixie was released.

--- a/.github/workflows/check.yml
+++ b/.github/workflows/check.yml
@@ -48,7 +48,7 @@ jobs:
         with:
           push: true
           tags: localhost:5000/name/app:latest
-      - name: lacheck
+      - name: squeeze
         working-directory: test
         run: |
           docker run -v $(pwd):/workdir localhost:5000/name/app:latest example.pdf example-small.pdf

--- a/.github/workflows/check.yml
+++ b/.github/workflows/check.yml
@@ -26,7 +26,7 @@ jobs:
     name: v1
     steps:
       - name: Checkout
-        uses: actions/checkout@v4
+        uses: actions/checkout@v5
       - uses: koppor/cpdfsqueeze-action@v1
         with:
           input_file: 'test/example.pdf'
@@ -36,7 +36,7 @@ jobs:
     name: latest
     steps:
       - name: Checkout
-        uses: actions/checkout@v4
+        uses: actions/checkout@v5
       - uses: koppor/cpdfsqueeze-action@latest
         with:
           input_file: 'test/example.pdf'
@@ -46,16 +46,8 @@ jobs:
     name: edge
     steps:
       - name: Checkout
-        uses: actions/checkout@v4
+        uses: actions/checkout@v5
       - uses: koppor/cpdfsqueeze-action@edge
         with:
           input_file: 'test/example.pdf'
           output_file: 'test/example-small.pdf'
-  keepalive:
-    name: Keepalive
-    runs-on: ubuntu-latest
-    permissions:
-      actions: write
-    steps:
-      - uses: actions/checkout@v4
-      - uses: gautamkrishnar/keepalive-workflow@v2

--- a/.github/workflows/check.yml
+++ b/.github/workflows/check.yml
@@ -53,7 +53,7 @@ jobs:
         run: |
           docker run -v $(pwd):/workdir localhost:5000/name/app:latest example.pdf example-small.pdf
       - name: Archive PDF
-        uses: actions/upload-artifact@v5
+        uses: actions/upload-artifact@v4
         with:
           name: example-small.pdf
           path: test/example-small.pdf

--- a/.github/workflows/check.yml
+++ b/.github/workflows/check.yml
@@ -21,8 +21,9 @@ permissions:
   contents: write
 
 jobs:
-  check:
+  check_HEAD:
     runs-on: ubuntu-latest
+    name: HEAD
     services:
       registry:
         image: registry:2

--- a/.github/workflows/check.yml
+++ b/.github/workflows/check.yml
@@ -57,6 +57,8 @@ jobs:
         with:
           name: example-small.pdf
           path: test/example-small.pdf
+
+  # One cannot use matrix at action references; therefore explicit enumeration
   check_v1:
     runs-on: ubuntu-latest
     name: v1

--- a/.github/workflows/check.yml
+++ b/.github/workflows/check.yml
@@ -57,17 +57,33 @@ jobs:
         with:
           name: example-small.pdf
           path: test/example-small.pdf
-  check-tags:
+  check_v1:
     runs-on: ubuntu-latest
-    strategy:
-      matrix:
-        version: [v1, latest, edge]
-    name: ${{ matrix.version }}
+    name: v1
     steps:
       - name: Checkout
         uses: actions/checkout@v5
-      - name: Run cpdfsqueeze ${{ matrix.version }}
-        uses: koppor/cpdfsqueeze-action@${{ matrix.version }}
+      - uses: koppor/cpdfsqueeze-action@v1
+        with:
+          input_file: 'test/example.pdf'
+          output_file: 'test/example-small.pdf'
+  check_latest:
+    runs-on: ubuntu-latest
+    name: latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v5
+      - uses: koppor/cpdfsqueeze-action@latest
+        with:
+          input_file: 'test/example.pdf'
+          output_file: 'test/example-small.pdf'
+  check_edge:
+    runs-on: ubuntu-latest
+    name: edge
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v5
+      - uses: koppor/cpdfsqueeze-action@edge
         with:
           input_file: 'test/example.pdf'
           output_file: 'test/example-small.pdf'

--- a/.github/workflows/check.yml
+++ b/.github/workflows/check.yml
@@ -21,6 +21,42 @@ permissions:
   contents: write
 
 jobs:
+  check:
+    runs-on: ubuntu-latest
+    services:
+      registry:
+        image: registry:2
+        ports:
+          - 5000:5000
+    steps:
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v2
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+        with:
+          driver-opts: network=host
+      - name: Cache Docker layers
+        uses: actions/cache@v3
+        with:
+          path: /tmp/.buildx-cache
+          key: ${{ runner.os }}-buildx
+      - name: Set up Git repository
+        uses: actions/checkout@v5
+      - name: Build
+        id: docker_build
+        uses: docker/build-push-action@v4
+        with:
+          push: true
+          tags: localhost:5000/name/app:latest
+      - name: lacheck
+        working-directory: test
+        run: |
+          docker run -v $(pwd):/workdir localhost:5000/name/app:latest example.pdf example-small.pdf
+      - name: Archive PDF
+        uses: actions/upload-artifact@v5
+        with:
+          name: example-small.pdf
+          path: test/example-small.pdf
   check_v1:
     runs-on: ubuntu-latest
     name: v1

--- a/.github/workflows/check.yml
+++ b/.github/workflows/check.yml
@@ -57,33 +57,17 @@ jobs:
         with:
           name: example-small.pdf
           path: test/example-small.pdf
-  check_v1:
+  check-tags:
     runs-on: ubuntu-latest
-    name: v1
+    strategy:
+      matrix:
+        version: [v1, latest, edge]
+    name: ${{ matrix.version }}
     steps:
       - name: Checkout
         uses: actions/checkout@v5
-      - uses: koppor/cpdfsqueeze-action@v1
-        with:
-          input_file: 'test/example.pdf'
-          output_file: 'test/example-small.pdf'
-  check_latest:
-    runs-on: ubuntu-latest
-    name: latest
-    steps:
-      - name: Checkout
-        uses: actions/checkout@v5
-      - uses: koppor/cpdfsqueeze-action@latest
-        with:
-          input_file: 'test/example.pdf'
-          output_file: 'test/example-small.pdf'
-  check_edge:
-    runs-on: ubuntu-latest
-    name: edge
-    steps:
-      - name: Checkout
-        uses: actions/checkout@v5
-      - uses: koppor/cpdfsqueeze-action@edge
+      - name: Run cpdfsqueeze ${{ matrix.version }}
+        uses: koppor/cpdfsqueeze-action@${{ matrix.version }}
         with:
           input_file: 'test/example.pdf'
           output_file: 'test/example-small.pdf'

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -5,6 +5,20 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [1.0.2] - 2025-08-14
+
+### Added
+
+- `entrypoint` now also used in the `Dockerfile` itself.
+
+### Fixed
+
+- Pin debian release to "bookworm". [#5](https://github.com/koppor/cpdfsqueeze-action/issues/5)
+
+### Changed
+
+- Using "$@" for parameter parsing [#6](https://github.com/koppor/cpdfsqueeze-action/pulls/6)
+
 ## [1.0.1] - 2024-03-11
 
 ### Fixed
@@ -15,5 +29,6 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 Initial public release
 
-[Unreleased]: https://github.com/koppor/cpdfsqueeze-action/compare/1.0.1...edge
+[Unreleased]: https://github.com/koppor/cpdfsqueeze-action/compare/1.0.2...edge
+[1.0.2]: https://github.com/koppor/cpdfsqueeze-action/compare/1.0.1...1.0.2
 [1.0.1]: https://github.com/koppor/cpdfsqueeze-action/compare/1.0.0...1.0.1

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -13,7 +13,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Fixed
 
-- Pin debian release to "bookworm". [#5](https://github.com/koppor/cpdfsqueeze-action/issues/5)
+- Pin debian release (for building) to "bookworm". [#5](https://github.com/koppor/cpdfsqueeze-action/issues/5)
 
 ### Changed
 

--- a/Dockerfile
+++ b/Dockerfile
@@ -15,3 +15,4 @@ COPY entrypoint.sh /entrypoint.sh
 RUN mkdir /workdir
 WORKDIR /workdir
 CMD ["cpdfsqueeze", "--help"]
+ENTRYPOINT ["/entrypoint.sh"]

--- a/Dockerfile
+++ b/Dockerfile
@@ -1,4 +1,4 @@
-FROM debian:stable-slim AS BUILDER
+FROM debian:bookworm-slim AS BUILDER
 ENV LANG=C.UTF-8 \
     LC_ALL=C.UTF-8 \
     TERM=dumb

--- a/entrypoint.sh
+++ b/entrypoint.sh
@@ -1,3 +1,3 @@
 #!/bin/sh
 set -e
-cpdfsqueeze $*
+cpdfsqueeze "$@"

